code/processes/gateway.q

```diff
@@ -409,6 +413,12 @@ getserverids:{[att]
   ];
 
   // its a dictionary of attributes
+    // check if attribute types are correct types
+    correctAttTypes:`dates`servertypes`tables!14 11 11h;
```

Comment:
Concerned this won't recognise attributes passed in as atoms so might want to adjust the dictionary to allow for this.

Reply:
Use `abs` in here somewhere to account for lists and atoms?